prprocessor/__main__.py

```diff
@@ -282,13 +291,67 @@ async def update_redmine_on_issues(pull_request: Mapping, issues: Iterable[Issue
 
 @process_event_actions('pull_request', {'opened', 'ready_for_review', 'reopened', 'synchronize'})
 @process_webhook_payload
-async def on_pr_modified(*, pull_request: Mapping, **other) -> None:  # pylint: disable=unused-argument
-    await run_pull_request_check(pull_request)
+async def on_pr_modified(*, action: str, pull_request: Mapping, **_kw) -> None:
+    conclusion = await run_pull_request_check(pull_request)
+
+    # TODO: retrieve active labels
+    labels: set[Label] = set()
+
+    if action == 'opened':
+        labels.add(Label.NOT_YET_REVIEWED)
+
+    if action == 'synchronize' and Label.WAITING_ON_CONTRIBUTOR in labels:
+        labels.remove(Label.WAITING_ON_CONTRIBUTOR)
+        pr_has_review = False # TODO
+        if pr_has_review:
+            labels.add(Label.NEEDS_RE_REVIEW)
+
+    if conclusion == 'failure':
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+    elif conclusion == 'success':
+        redmine_required = False # TODO
+        valid_commit_style = False # TODO
+        if redmine_required and not valid_commit_style:
+            labels.add(Label.WAITING_ON_CONTRIBUTOR)
+        else:
+            labels.remove(Label.WAITING_ON_CONTRIBUTOR)
+
+    # TODO: handle None value and resubmit later?
+    if pull_request['mergable'] is False:
+        # TODO: post message?
+        labels.remove(Label.NEEDS_RE_REVIEW)
+        labels.remove(Label.NOT_YET_REVIEWED)
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+
+    # TODO: path labels
+    # TODO: branch labels
+
+    # TODO: apply labels
+
+
+@process_event_actions('pull_request_review', {'submitted'})
+@process_webhook_payload
+async def on_pr_review_assign_labels(*, action: str, review: Mapping, **_kw) -> None:
+    # TODO: retrieve active labels
+    labels: set[Label] = set()
+
+    # TODO: look at review['author_association']?
```

Comment:
This isn't in the legacy app, but something I noticed while reading the docs. Today anyone can review a PR, which may not be great.